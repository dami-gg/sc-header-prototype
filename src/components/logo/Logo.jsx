import React, {Component} from 'react';
import logo from '../../assets/img/logo.svg';
// import smallLogo from '../../assets/img/logo--no-text.svg';
import './Logo.scss';

class Logo extends Component {
  render() {
    return (
        <div className="dc-column dc-column--shrink logo-container"> {/* TODO Add onClick and go to Solution Center--> */ }
          <img src={logo}
               alt="Logo"
               className="logo logo--large logo--clickable" />
          { /*
          <img src={smallLogo}
               alt="Logo"
               className="logo logo--small logo--clickable" />
               */ }
        </div>
    );
  }
}

export default Logo;
