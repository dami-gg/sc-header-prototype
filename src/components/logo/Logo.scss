@import "../../../node_modules/dress-code/dist/sass/dress-code";

.logo-container {
  display: flex;
  max-height: 2rem;

  &--desktop {
    display: none;

    @media (min-width: $dc-huge-width) {
      display: block;
    }
  }

  &--mobile {
    margin-left: 0;

    @media (min-width: $dc-huge-width) {
      display: none;
    }
  }

  .logo {
    cursor: pointer;

    img {
      max-height: $dc-touch-size - $dc-space150;
      max-width: 90px;
    }
  }

  .menu-toggle {
    @media (min-width: $dc-huge-width) {
      display: none;
    }

    .dc-icon {
      margin-left: 4px;
      vertical-align: middle;
    }
  }
}




