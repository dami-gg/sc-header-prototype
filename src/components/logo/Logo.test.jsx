import React from 'react';
import renderer from 'react-test-renderer';
import Logo from './Logo';

describe('Logo component', () => {
  it('should render correctly', () => {
    let tree = renderer.create(
        <Logo
          source="source"
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});