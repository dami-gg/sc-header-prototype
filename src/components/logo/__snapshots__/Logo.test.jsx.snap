exports[`Logo component should render correctly 1`] = `
<div
  className="logo">
  <img
    alt="Logo"
    src="source" />
</div>
`;
